kong/runloop/handler.lua

```diff
@@ -78,13 +78,26 @@ do
     return false
   end
 
+
+  local function load_service_from_db(service_pk)
+    return kong.db.services:select(service_pk)
+  end
+
   local function get_service_for_route(db, route)
     local service_pk = route.service
     if not service_pk then
       return nil
     end
 
-    local service, err = db.services:select(service_pk)
+    local service, err
+    if kong.cache then -- the kong.cache module is sometimes missing during tests
```

Comment:
imho, tests should be explicit and require/mock a cache instance. This is a dangerous precedent being made here, in which production (proxy path) code is being tested differently during tests than it is at runtime.

Reply:
You are absolutely right about this one. I went back and removed the if, in order to do just that...

... and then I discovered that we create the initial router on `init` phase, but the cache is only available after `init_worker` phase. So on the first iteration this was still doing the 1000 db queries to the database.

I have changed the code so that it falls back to a regular lua table on init phase, and also left comments in the code explaining that the `if kong.cache ...` happens because of this, not because of tests, as I first thought.

Thanks for your feedback!

Reply:
Awesome! :100: